# Assignment 1

In [10]:
a= "Hello Python"
a= a[6:12] +" "+ a[0:5]
print a

Python Hello


slicing 연산자와 +를 이용해 쉽게 풀수 있었습니다.

In [12]:
b="Hello Python World"
b= b[13:18]+ " "+ b[6:13] + b[0:5]
print b

World Python Hello


1번 문제와 같이 slicing 연산자와 +를 이용해 쉽게 해결할수 있습니다.

In [14]:
c="Hello"
c= c[::-1]
print c

olleH


slicing 연산자에 뒤에 :를 하나추가하고 뒤에 -1을 써주면 역방향으로 출력되는것을 이용 하였습니다.

In [21]:
s="python"
print s[0] + s[0][0] + s[0][0][0]


ppp


 모두 제일 처음 문자인p가 출력됩니다. 제 생각에 [0]은 제일 처음을 가리키며 가리키는 곳에 문자가 없으면 그냥 오류 없이 자동으로 소멸되어 모두 s[0] 취급 하는 것 같습니다.

In [26]:
s="python"
print s[-100]

IndexError: string index out of range

뒤에서부터 앞으로 100번째 되는 문자를 가리켜 출력을 해야하는데 단어가 100보다 짧아 에러가 에러가나오는 것 같습니다.

In [27]:
s="python"
print s[100]

IndexError: string index out of range

마찬가지로 100번째 되는 되는문자를 가리켜야 하지만 그곳에 문자가 없기 때문에 에러가 발생한 것 같습니다.

In [33]:
s="python"
print s[-100, 100]

TypeError: string indices must be integers, not tuple

문자열은 출력하려면 하나의 정수를 써주어야 하는데 두개의 정수값을 써 에러가 발생한 것 같습니다.

In [34]:
s="python"
print s[1:-1]

ytho


slicing 연산자로 처음 start값은 1로 로y부터 시작하여 stop값이 -1로 -1은 가장 마지막을 가리켜 n을 가리키지만 stop의 특성상 그 앞의 문자에서
끊기기 때문에 o까지 출력 됩니다.

In [38]:
s="python"
print s[3:-3]

slicing 연산자로 처음 start값은 3으로 h부터 시작하여 stop값이 -3으로 마지막에서 3번째 앞인 h를 가리키게 되는데 stop의 특성상 그 앞의 문자인
t까지 출력되어야 하나 h와 t 는 지나쳐서 사이에 문자가 없으므로 출력되는 값이 없습니다.

In [50]:
for x in range(100):
    if x%2==0:
        continue
    print x,



1 3 5 7 9 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 51 53 55 57 59 61 63 65 67 69 71 73 75 77 79 81 83 85 87 89 91 93 95 97 99


for문으로 0부터 100까지 출력을 하되 포문 for문 다음에 if문으로 2로 나누어준 나머지가 0이되는 숫자는 출력이 되지 않도록 continue문을 써주어 
홀수만 출력 하였습니다.

In [65]:
a=2
sum=0;
while a <= 100:
    sum=sum+a
    a = a + 2
  
print sum



2550


while문을 사용하여 a값이 100과 같거나 작을때까지 출력하는데 a값은 0부터 2씩 커지도록 +2를 해주어 짝수만 출력되게 하였고 
중간에 sum변수에 a값을 모두 더해 주어 출력을 하니 2550 이라는 값이 나왔습니다.나왔습니다

In [ ]:
a =100
print a